File: Assignment1/Question2/Mat_Mul_Int.py
import time
import numpy as np

def matrix_multiply_int(N):
    A = np.ones((N, N), dtype=int)
    B = np.ones((N, N), dtype=int)
    C = np.zeros((N, N), dtype=int)

    start = time.time()

    for i in range(N):
        for j in range(N):
            for k in range(N):
                C[i][j] += A[i][k] * B[k][j]

    end = time.time()

    print(f"Time taken for N={N} (int): {end - start:.6f} s")

if __name__ == "__main__":
    matrix_multiply_int(1024) # Hardcode N here in sizes = [64, 128, 256, 512, 1024]

File: Assignment1/Question2/Mat_Mul_Int.cpp
#include <iostream>
#include <vector>
#include <ctime>
using namespace std;

void matrix_multiply_int(int N) {
    vector<vector<int> > A(N, vector<int>(N, 1));
    vector<vector<int> > B(N, vector<int>(N, 1));
    vector<vector<int> > C(N, vector<int>(N, 0));

    clock_t start = clock();

    for (int i = 0; i < N; ++i) {
        for (int j = 0; j < N; ++j) {
            for (int k = 0; k < N; ++k) {
                C[i][j] += A[i][k] * B[k][j];
            }
        }
    }

    clock_t end = clock();
    double diff = double(end - start) / CLOCKS_PER_SEC;

    cout << "Time taken for N=" << N << " (int): " << diff << " s\n";
}

int main() {
    matrix_multiply_int(1024); // Hardcode N here for [64, 128, 256, 512, 1024]
    return 0;
}
